backend/app/core/background_runner.py

```diff
@@ -0,0 +1,43 @@
+import asyncio
+
+from app.core.bot import BaseBotClient
+from app.core.config import settings
+from app.core.sessions import redis_session_storage
+
+
+class MatrixBotBackgroundRunner:
+    def __init__(self):
+        self.client = BaseBotClient(homeserver=settings.matrix_bot.homeserver)
+        self.background_task = None
+
+    async def start_bot(self):
+        try:
+            # Fetch next batch token stored in redis
+            self.client.next_batch = redis_session_storage["next_batch_token"]
```

Comment:
What's the behaviour here if there isn't a next batch in redis?

Reply:
https://github.com/MatrixDoctors/matrix-cerberus/blob/e6897e10164c15c7986d9cf57ff0af3f99a5c5c6/backend/app/core/sessions.py#L25-L27

It returns `None` as mentioned here https://redis-py.readthedocs.io/en/stable/commands.html#redis.commands.core.CoreCommands.get

Reply:
I'll mention it as a comment.

Reply:
Cool that's fine.